File: python-wrapper/src/python/model/embed/DocumentStatus.py
from typing import Union, Dict
from enum import Enum

class DocumentStatus(Enum):
	Draft = "draft"
	Finalized = "finalized"
	PendingReview = "pending_review"
	Reviewed = "reviewed"
	PendingSignature = "pending_signature"
	Signed = "signed"
	Canceled = "canceled"
	Sent = "sent"
	Delivered = "delivered"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'DocumentStatus':
		if data == "draft":
			return DocumentStatus.Draft
		elif data == "finalized":
			return DocumentStatus.Finalized
		elif data == "pending_review":
			return DocumentStatus.PendingReview
		elif data == "reviewed":
			return DocumentStatus.Reviewed
		elif data == "pending_signature":
			return DocumentStatus.PendingSignature
		elif data == "signed":
			return DocumentStatus.Signed
		elif data == "canceled":
			return DocumentStatus.Canceled
		elif data == "sent":
			return DocumentStatus.Sent
		elif data == "delivered":
			return DocumentStatus.Delivered
		else:
			raise Exception(f"{data} is not a valid value for DocumentStatus enum.")

__all__ = ['DocumentStatus']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/DocumentStatus.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class DocumentStatus(
	internal val dtoSerialName: String,
) {
	@SerialName("draft")
	Draft("draft"),

	@SerialName("finalized")
	Finalized("finalized"),

	@SerialName("pending_review")
	PendingReview("pending_review"),

	@SerialName("reviewed")
	Reviewed("reviewed"),

	@SerialName("pending_signature")
	PendingSignature("pending_signature"),

	@SerialName("signed")
	Signed("signed"),

	@SerialName("canceled")
	Canceled("canceled"),

	@SerialName("sent")
	Sent("sent"),

	@SerialName("delivered")
	Delivered("delivered"),
}
